src/platform/forms-system/src/js/components/Telephone.jsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+
+// import recordEvent from 'platform/monitoring/record-event.js';
+
+const knownNumbers = {
+  '911': ['911'],
+  '8002738255': ['crisisline', 'crisis'],
+  '8003331795': ['usdirect', 'direct', 'express'],
+  '8005351117': ['ncacenter', 'nca'],
+  '8005389552': ['dslogon', 'dmdc'],
+  '8007994889': ['vetcrisistty', 'crisistty'],
+  '8008271000': ['hotline', 'vabenefits', 'benefits', '1000'],
+  '8008778339': ['helptty'],
+  '8446982311': ['va311', '311'],
+  '8552603274': ['caregiver'],
+  '8555747286': ['hrc', 'reportline'],
+  '8663632883': ['dslogontty', 'dmdctty'],
+  '8772228387': ['helpline', 'help', '222vets'],
+  '8773270022': ['mhv', 'health', 'evet'],
+  '8774243838': ['4aidvet', '4aid', 'homeless'],
+  '8884424551': ['gibill', 'edu'],
+};
+
+const patterns = {
+  '911': '###',
+  default: '###-###-####',
+};
+
+const recordClick = () => {
+  // TO-DO: pass event name & location URL
+  // recordEvent({ event: `telephone-click` });
+};
+
+// Allow labels "gi-bill", "GI_Bill" or "GI BILL"
+const formatLabel = label => label?.replace(/[-_\s]/g, '').toLowerCase();
+
+// Strip out leading "1" and any non-digits
+const cleanHref = number =>
+  number
+    .replace(/^1/, '') // strip leading "1" from telephone number
+    .replace(/[^\d]/g, '');
+
+// Create link text from pattern
+const formatTelText = (num, pattern) => {
+  let i = 0;
+  return pattern.replace(/#/g, () => num[i++] || '');
+};
+
+const formatBlock = number => {
+  const len = number.length - 1;
+  // return rounded number as a grouped value, e.g. `800` or `1000`
+  // but split non-round numbers, e.g. `827` => `8 2 7`
+  const roundNumber = parseInt(number.slice(-len), 10) === 0;
+  return roundNumber ? number : number.split('').join(' ');
+};
+
+// Combine phone number blocks within the label separated by periods
+// "800-555-1212" => "800. 5 5 5. 1 2 1 2"
+const formatTelLabel = number =>
+  number
+    .split(/[^\d]+/)
```

Comment:
You could use `Array.from(number)` to convert to an array

Reply:
Hmm, I think we need `split` because it needs to split on non-digits. If a number like `(800) 555-1212` was passed to that function, it'd need to remove those non-digit characters from the array. I guess I could filter them out in the next step.

Reply:
Ah okay, that makes sense!

Reply:
If it makes it easier, you could have the phone number passed as an object with an areacode, number, etc to make it easier to work with instead of a string. If the numbers are centralized into `CONTACTS`, then the format of the data is up to you

Reply:
having an object with the number, and trying to apply that to the pattern would be more complicated, I think. This seems pretty simple & straight-forward. I'm also trying to minimize what you need to type when you add props to the component; we're all a little bit lazy when it comes to typing 😉